api/client/client.go

```diff
@@ -59,34 +56,16 @@ func NewServer(addr string) Remote {
 }
 
 func (srv *server) Hosts() []string {
-	return []string{net.JoinHostPort(srv.Address, fmt.Sprintf("%d", srv.Port))}
+	return []string{srv.URL}
 }
 
 func newServer(addr string) *server {
-	addr = strings.TrimSpace(addr)
-	host, port, err := net.SplitHostPort(addr)
-	if err != nil {
-		host, port, err = net.SplitHostPort(addr + ":8888")
-		if err != nil {
-			return nil
-		}
-	}
-
-	var portno int
-	if port == "" {
-		portno = 8888
-	} else {
-		portno, err = strconv.Atoi(port)
-		if err != nil {
-			return nil
-		}
-	}
 
-	return &server{host, portno}
+	return &server{addr}
```

Comment:
Existing clients use hosts that do not contain the protocol version. You should add http:// as a default if none is defined.